utils/replicaset/replicaset.go

```diff
@@ -57,6 +64,69 @@ func FindNewReplicaSet(rollout *v1alpha1.Rollout, rsList []*appsv1.ReplicaSet) *
+			RSTemplate.Spec.Affinity = &corev1.Affinity{}
+		}
+		if RSTemplate.Spec.Affinity.PodAntiAffinity == nil {
+			RSTemplate.Spec.Affinity.PodAntiAffinity = &corev1.PodAntiAffinity{}
+		}
+		if RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+			RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = []corev1.PodAffinityTerm{antiAffinityRule}
+		} else {
+			exists := CheckIfDefaultAntiAffinityRuleExists(RSTemplate, rollout)
+			if exists == -1 {
+				RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = append(RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution, antiAffinityRule)
+			}
+		}
+	}
+}
+
+func GetDefaultAntiAffinityRule(rollout v1alpha1.Rollout) corev1.PodAffinityTerm {
+	// Create anti-affinity rule for last stable rollout
+	antiAffinityRule := corev1.PodAffinityTerm{
+		LabelSelector: &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{{
+				Key:      v1alpha1.DefaultRolloutUniqueLabelKey,
+				Operator: metav1.LabelSelectorOpIn,
+				// Most recent stable ReplicaSet
+				Values: []string{rollout.Status.StableRS},
+			}},
+		},
+		Namespaces:  []string{rollout.Namespace},
+		TopologyKey: "kubernetes.io/hostname",
+	}
+	return antiAffinityRule
+}
+
+func CheckIfDefaultAntiAffinityRuleExists(RSTemplate *corev1.PodTemplateSpec, rollout v1alpha1.Rollout) int {
+	defaultAntiAffinityRule := GetDefaultAntiAffinityRule(rollout)
+	if RSTemplate.Spec.Affinity != nil && RSTemplate.Spec.Affinity.PodAntiAffinity != nil && RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+		for i, podAffinityTerm := range RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution {
+			if cmp.Equal(podAffinityTerm, defaultAntiAffinityRule) {
+				return i
+			}
+		}
+	}
+	return -1
+}
+
+func RemoveDefaultAntiAffinityRuleIfExists(RSTemplate *corev1.PodTemplateSpec, rollout v1alpha1.Rollout) {
+	i := CheckIfDefaultAntiAffinityRuleExists(RSTemplate, rollout)
+	if i >= 0 {
+		RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = append(RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution[:i], RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution[i+1:]...)
+		if RSTemplate.Spec.Affinity.PodAntiAffinity == (&corev1.PodAntiAffinity{}) { //RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil && RSTemplate.Spec.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution == nil {
+			RSTemplate.Spec.Affinity.PodAntiAffinity = nil
+		}
+		if RSTemplate.Spec.Affinity == (&corev1.Affinity{}) { //RSTemplate.Spec.Affinity.PodAntiAffinity == nil && RSTemplate.Spec.Affinity.PodAffinity == nil && RSTemplate.Spec.Affinity.NodeAffinity == nil {
```

Comment:
I have a strong feeling that this if statement will never evaluate to true since you are comparing two pointers and not their values. This would be a place to use the `reflect.DeepEqual` function since that will ensure that the two structs have the same values.